src/Compilers/CSharp/Portable/Symbols/Source/LambdaSymbol.cs

```diff
@@ -360,6 +367,11 @@ public override bool HidesBaseMethodsByName
                     refKind = RefKind.None;
                 }
 
+                if (refKind == RefKind.RefReadOnly)
+                {
+                    compilation.EnsureIsReadOnlyAttributeExists(diagnostics, unboundLambda.ParameterLocation(p), synthesizeEmbeddedAttributes: false);
+                }
+
                 var name = unboundLambda.ParameterName(p);
                 var location = unboundLambda.ParameterLocation(p);
```

Comment:
Perhaps move `var location` ahead of `if` above and use `location` in `EnsureIsReadOnlyAttributeExists` call.